piston/configuration/config_loader.py

```diff
@@ -0,0 +1,78 @@
+import os
+import platform
+from typing import Optional
+
+import yaml
+from piston.configuration.config_validator import validate_config
+from piston.utilities.constants import Configuration
+from rich.console import Console
+
+
+class ConfigLoader:
+    """Loads yaml config files to customize piston-cli."""
+
+    def __init__(self, path: Optional[str]):
+        self.console = Console()
+        self._path = Configuration.configuration_paths[platform.system()]
+        if path is not None:
+            self.set_path(path)
+        self._config = {}
+
+    def set_path(self, path: str) -> None:
+        """Sets the path of private variable _path."""
+        self._path = path
+
+    def _load_yaml(self) -> None:
+        """Loads the keys and values from a yaml file."""
+        expanded_path = os.path.abspath(os.path.expandvars(self._path))
+
+        self.console.print(f"[green]Loading config:[/green] {expanded_path}")
+
+        with open(expanded_path) as loaded_config:
+            loaded_config = yaml.load(loaded_config, Loader=yaml.FullLoader)
+
+        for key, value in loaded_config.items():
+            if key in Configuration.default_configuration:
+                self._config[key] = value
+                self.console.print(f"[green]- Loaded {key}: {value}[/green]")
```

Comment:
Currently it fails when the use adds `-` in front of the theme name like:
```yaml
theme:
        - dracula
```
But this is very common in `yaml` files, so we can add support for it by checking if it is a `list` and its length is exactly `1`

```suggestion
                if isinstance(value, list) and len(value) == 1:
                    self._config[key] = value[0]
                else:
                    self._config[key] = value

                self.console.print(f"[green]- Loaded {key}: {self._config[key]}[/green]")
```

Reply:
Another possibility is:
If the `theme` config is a list, it would randomly select from it.

Reply:
I think we discussed this in Discord, I'll opt for the second route because I can just check if it is a list, and if it is, notify the user that it will be picked from randomly, even if it is only one theme, it will make more sense than just letting lists of length one act differently, I think.

Reply:
> I think we discussed this in Discord, I'll opt for the second route because I can just check if it is a list, and if it is, notify the user that it will be picked from randomly, even if it is only one theme, it will make more sense than just letting lists of length one act differently, I think.

Right, go ahead with the second route.